extensions/functions_arithmetic.yaml

```diff
@@ -1614,3 +1614,131 @@ window_functions:
+    description: >
+      The `expression` is evaluated against the first row.  The `expression` can be a
+      column, expression or subquery that evaluates to a single value.
+    impls:
+      - args:
+          - value: any1
+            name: expression
+        nullability: DECLARED_OUTPUT
+        decomposable: NONE
+        return: any1?
+        window_type: PARTITION
+  - name: "last_value"
+    description: >
+      The `expression` is evaluated against the last row.  The `expression` can be a
+      column, expression, or subquery that evaluates to a single value.
+    impls:
+      - args:
+          - value: any1
+            name: expression
+        nullability: DECLARED_OUTPUT
+        decomposable: NONE
+        return: any1?
+        window_type: PARTITION
+  - name: "nth_value"
+    description: >
+      The `expression` is evaluated against a row that comes after the current row based
+      on the `offset`.  `offset` should be a positive integer.  The `expression` can be a
+      column, expression or subquery that evaluates to a single value.
+    impls:
+      - args:
+          - value: any1
+            name: expression
+          - value: i32
+            name: offset
+        nullability: DECLARED_OUTPUT
+        decomposable: NONE
+        return: any1?
+        window_type: PARTITION
+      - args:
+          - value: any1
+            name: expression
+          - value: i64
+            name: offset
+        nullability: DECLARED_OUTPUT
+        decomposable: NONE
+        return: any1?
+        window_type: PARTITION
+  - name: "lead"
+    description: >
+      Return a value from a following row based on a specified physical offset.
+      This allows you to compare a value in the current row against a following row.
+
+      The `expression` is evaluated against a row that comes after the current row based
+      on the `offset`.  The `expression` can be a column, expression or subquery that
+      evaluates to a single value. The `offset` should be a positive integer and is set to
+      1 if not specified explicitly. The function returns the `default` input value if
+      `offset` goes beyond the scope of the partition. If a `default` value is not
+      specified, it is set to `null`
+
+      Example comparing the sales of the current year to the following year.
+      | year | sales  | next_year_sales |
+      | 2019 | 20.50  | 30.00           |
+      | 2020 | 30.00  | 45.99           |
+      | 2021 | 45.99  | null            |
+    impls:
+      - args:
+          - value: any1
+            name: expression
+          - value: i32
+            name: offset
+          - value: any1
+            name: default
+        nullability: DECLARED_OUTPUT
+        decomposable: NONE
+        return: any1?
+        window_type: PARTITION
+      - args:
+          - value: any1
+            name: expression
+          - value: i64
+            name: offset
+          - value: any1
+            name: default
```

Comment:
I'm not certain there is much value in an `i64` variant.  I don't think there are any realistic scenarios where a window has more than 2^32 rows.  We can always add that later if needed.  I suppose this same logic applies to lag and lead.